Develop logs

Aldatrualda opened the pull request.

Commit:
+ Added logs
+ Added Show Employee Page
+ Added Show Player Page
+ Added Account class

Commit:
- Account class
- MainPageCoach

Commit:
# Depends on who logs in Coach or Admin, opens the main page with restricted abilities whether it's a coach.

DenisPushka commented on RealMadridDesktopApplication/Forms/LoginPage.cs:
```diff
@@ -1,13 +1,19 @@
         {
+            logger.Info("Login Page was opened");
             InitializeComponent();
+            logger.Info("Login Page was closed");
```
Разве окно здесь закрывается?)

Aldatrualda replied:
Ну оно будет закрываться, ахаха. Я настрою это :)

DenisPushka commented on RealMadridDesktopApplication/Forms/LoginPage.cs:
```diff
@@ -16,57 +22,90 @@ private void buttonClear_Click(object sender, EventArgs e)
-            return "SELECT * FROM employee_of_real_madrid WHERE login = '" + textBoxLogin.Text + "'AND password = '" + textBoxPassword.Text + "'";
+            string login;
+            string password;
+            NpgsqlDataAdapter npgsqlDataAdapter = new NpgsqlDataAdapter(selectQuery, connection);
+
+            DataTable dataTable = new DataTable();
+            npgsqlDataAdapter.Fill(dataTable);
+
+            if (dataTable.Rows.Count > 0)
```
Вот здесь уж лучше прокидывать запрос на получение количества записей, а не всю выборку

DenisPushka commented on RealMadridDesktopApplication/Forms/LoginPage.cs:
```diff
@@ -16,57 +22,90 @@ private void buttonClear_Click(object sender, EventArgs e)
         }
 
-        private NpgsqlConnection GetConnection()
+        private string GetAccessModifierOfAccount(string login, string password)
```
Должен возвращать enum, в случае ошибки - null

Aldatrualda replied:
Исправил. 
Еще в лог добавил переменную accessModifier в случае ошибки.

![Uploading image.png…]()

DenisPushka replied:
и где исправление?

Aldatrualda replied:
![image](https://github.com/Aldatrualda/RealMadridDesktopApplication/assets/112622972/2fb42f06-7a36-4fbc-8242-2fd7de3f4e22)

DenisPushka commented on RealMadridDesktopApplication/Modules/Player.cs:
```diff
@@ -11,11 +11,12 @@ public class Player : PersonalDetails
             this.natinality = natinality;
             this.address = address;
             this.location = location;
+            Birthday = birthdayBuffer[2] + "-" + birthdayBuffer[1] + "-" + birthdayBuffer[0];
```
магические цифры)

Aldatrualda replied:
Никакой магии только ловкость рук. Кстати, исправил в классе Employee переменные, там ошибка в написании была. В слове BIRTHDAY была ошибка.

![image](https://github.com/Aldatrualda/RealMadridDesktopApplication/assets/112622972/79c65ffe-0cda-4f04-bd49-fd1be740b838)

DenisPushka commented on RealMadridDesktopApplication/Password/GeneratePassword.cs:
```diff
@@ -0,0 +1,35 @@
+
+        public GeneratePassword() => this.password = GenerateRandomPassword(listOfChars);
+        
+        private string GenerateRandomPassword(string listOfChars)
```
добавь коммент, как происходит генерация

Aldatrualda replied:
Добавил.
![image](https://github.com/Aldatrualda/RealMadridDesktopApplication/assets/112622972/f7e58975-5819-4b36-9f94-58c2ed093990)

DenisPushka commented on RealMadridDesktopApplication/SQLConnection/SQLVariableContainer.cs:
```diff
@@ -0,0 +1,59 @@
+        /// <summary>
+        /// The info about the local database. I use it to connect to my SQL server.
+        /// </summary>
+        public static string ConnectionToSQL = "Host=localhost;Port=5432;Database=RealMadridDB;Username=postgres;Password=123456";
```
Попробуй вынести в зависимости проекта

DenisPushka replied:
Хорошо, оставим на следующую задачку. Делаем не в этой

DenisPushka reviewed (commented):
Уже очень хорошо выглядит, но есть небольшие недочеты.

Aldatrualda commented:
переместил NLog.config в отдельную папку.

![image](https://github.com/Aldatrualda/RealMadridDesktopApplication/assets/112622972/ba7ae307-17a3-4f1f-8f42-0ed21872a940)

Commit:
# Changed the package name of the GeneratePassword class to Service
# Corrected names of Birthday variables for in Employee class
# Changed a SQL query. Now we get a count to authorize an user
+ Added comments and variable for Birhday in Player class
+ Added configutaions pasckage

Commit:
+ Added Microsoft.Extensions.Configuration

DenisPushka commented on RealMadridDesktopApplication/Forms/LoginPage.cs:
```diff
@@ -16,57 +22,100 @@ private void buttonClear_Click(object sender, EventArgs e)
+            using (NpgsqlCommand command = new NpgsqlCommand(countQuery, connection))
+            {
+                NpgsqlDataReader dataReader = command.ExecuteReader();
+                if (dataReader.Read())
+                {
+                    employeeCount = dataReader.GetInt32(0);
+                }
+                else
+                {
+                    employeeCount = 0;
```
Лишнее, у тебя итак изначально переменная равна 0. Если уж хочешь заморочиться с присвоением, то можно было сразу тернарный оператор использовать. Если будешь его использовать, то убери создание в 50 строке, и сделай, что бы все происходило при создании, те: var buffer = logic ? 1: 0;

Aldatrualda replied:
Я исправил. 

![image](https://github.com/Aldatrualda/RealMadridDesktopApplication/assets/112622972/d182263c-0932-45c3-819f-9497c63ef3d4)

Aldatrualda replied:
Но мне кажется, что бред ...

Commit:
# in the method GetAccessModifierOfAccount returns AccessModifier insted of string
# replaced if statements in GetAccessModifierOfAccount with the ternary conditional operator

- removed login and password variables
# replaced int employeeCount with bool employeeLogin
